spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe 'Customer Show Page', type: :feature do
+ describe 'As a Visitor' do
+  before(:each) do
+    @customer_isaac = Customer.create!(name: "Isaac")
+    @customer_jack = Customer.create!(name: "Jack")
+    @customer_noah = Customer.create!(name: "Noah")
+
+    @walmart = Supermarket.create!(name: "Walmart Summit", location: "150 Summit Blvd")
+    @costco = Supermarket.create!(name: "Costco", location: "20 Marshall Rd")
+    @safeway = Supermarket.create!(name: "Safeway", location: "455 Coalton Rd")
+
+    @milk = @walmart.items.create!(name: "Milk", price: 2)
+    @banana = @walmart.items.create!(name: "Banana", price: 1)
+    @apple = @costco.items.create!(name: "Apple", price: 5)
+    @popcorn = @costco.items.create!(name: "Pop Corn", price: 8)
+    @mushroom = @safeway.items.create!(name: "Mushroom", price: 1)
+    @kiwi = @safeway.items.create!(name: "Kiwi", price: 3)
+
+    @customer_isaac.items << @milk
+    @customer_isaac.items << @popcorn
+    @customer_noah.items << @apple
+    @customer_noah.items << @kiwi
+    @customer_jack.items << @mushroom
+    @customer_jack.items << @banana
```

Comment:
Just a quick tip, if you want to reduce the amount you have to type when associating many things like you have on lines 21-26 you can use this syntax:

```ruby
@customer_isaac.items << @milk << @popcorn
@customer_noah.items << @apple << @kiwi
@customer_jack.items << @mushroom << @banana
```

You can also reduce the amount you have to type if you place your `visit` in your `before :each` as well.